framework/branch-key-store-admin.md

```diff
@@ -0,0 +1,185 @@
+one designated for Encrypt.
+
+This allows MPL Consumers to configure different
+credentials or request headers for the KMS Operations.
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Initialization
+
+The Key Store Admin requires the following arguments:
+
+- [Storage](../branch-key-store.md#storage)
+- [Logical Key Store Name](../branch-key-store.md##logical-keystore-name)
+
+### User Agent
+
+Whenever possible,
+the Key Store Admin SHOULD
+append a user agent string to AWS Service requests with
+the value of `aws-kms-hierarchy`.
+
+## Operations
+
+The Key Store Admin MUST support the following operations:
+
+- [CreateKey](#createkey)
+- [VersionKey](#versionkey)
+
+### CreateKey
+
+The CreateKey caller MUST provide:
+
+- A [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn)
+
+The CreateKey caller MAY provide:
+
+- An optional Identifier
+- An optional Encryption Context
+- An optional [Key Management Strategy](#key-management-strategy)
+
+At this time,
+the [Key Management Strategy](#key-management-strategy) MUST be `AwsKmsReEncrypt`.
+
+The Operation behaves identically to the [Key Store Client's CreateKey](./branch-key-store.md#createkey),
+with the following caveats:
+
+#### KMS Configuration
+
+Where ever the Key Store Client's CreateKey method refers to the Key Store's KMS Configuration,
+use the equivalent [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn) expression.
+
+#### [Branch Key and Beacon Key Creation](./branch-key-store.md#branch-key-and-beacon-key-creation)
+
+The `AwsKmsReEncrypt` configuration MUST be treated as if it were the Key Store's `AwsKms`.
+
+### VersionKey
+
+The VersionKey caller MUST provide:
+
+- A [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn)
+- A `Identifier`
+
+The VersionKey caller MAY provide:
+
+- An optional [Key Management Strategy](#key-management-strategy)
+
+At this time,
+the [Key Management Strategy](#key-management-strategy) MUST be `AwsKmsReEncrypt`.
+
+The Operation behaves identically to the [Key Store Client's VersionKey](./branch-key-store.md#versionkey),
+with the following caveats:
+
+#### KMS Configuration
+
+Where ever the Key Store Client's CreateKey method refers to the Key Store Client's KMS Configuration,
+use the equivalent [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn) expression.
+
+#### [Wrapped Branch Key Creation](./branch-key-store.md#wrapped-branch-key-creation)
+
+The `AwsKmsReEncrypt` configuration MUST be treated as if it were the Key Store's `AwsKms`.
+
+<!--  LocalWords:  MRK AwsKms grantTokenList kmsClient ReEncrypt  -->
+<!--  LocalWords:  AwsKmsReEncrypt keystore AwsKmsDecryptEncrypt  -->
+<!--  LocalWords:  Admin ReEncrypt Changelog aws arn createkey -->
+<!--  LocalWords:  AwsCryptographyKeyStoreOperations versionkey GenerateDataKeyWithoutPlaintext -->
```

Comment:
```suggestion
```

Reply:
If this is not blocking, I would like to keep these here;
I will need to write the Mutation Spec,
and these spelling hints will help me.